cpp/roaring64map.hh

```diff
@@ -173,11 +173,11 @@ class Roaring64Map{
      * modified.
      */
     Roaring64Map &operator&=(const Roaring64Map &r) {
-        for (const auto& map_entry : r.roarings) {
-            if (roarings.count(map_entry.first) == 1) {
-                roarings[map_entry.first] &= map_entry.second;
-                roarings[map_entry.first].setCopyOnWrite(copyOnWrite);
-            }
```

Comment:
For the and operation: the left hand side map was erroneously holding on to Roarings representing ranges that weren't present in the right hand side in the previous implementation.